### PR TITLE
autopep8

### PR DESCRIPTION
**Pull request by mlautman**
Results of running 
`autopep8 --max-line-length 130 --in-place $1` on each `.py` file

### CODE REVIEW

File: src/rqt_launch/launch_widget.py
Comment (rsinnet) [inline src/rqt_launch/launch_widget.py]: Probably makes sense to remove some of the spaces on the right side of the comment hash marks.

File: src/rqt_launch/launch_widget.py
Comment (brawner) [inline src/rqt_launch/launch_widget.py]: Align with str.format

File: src/rqt_launch/launch_widget.py
Comment (brawner) [inline src/rqt_launch/launch_widget.py]: Is this commented code needed?

Comment (brawner) [reply src/rqt_launch/launch_widget.py]: Is this method needed?

File: src/rqt_launch/node_controller.py
Comment (brawner) [inline src/rqt_launch/node_controller.py]: Does this line need to be broken?

File: src/rqt_launch/node_controller.py
Comment (brawner) [inline src/rqt_launch/node_controller.py]: Does this line need to be broken?

File: src/rqt_launch/node_controller.py
Comment (brawner) [inline src/rqt_launch/node_controller.py]: This line doesn't need to be broken

File: src/rqt_launch/node_controller.py
Comment (brawner) [inline src/rqt_launch/node_controller.py]: This line doesn't need to be broken

File: src/rqt_launch/node_controller.py
Comment (brawner) [inline src/rqt_launch/node_controller.py]: This line doesn't need to be broken

File: src/rqt_launch/node_delegate.py
Comment (brawner) [inline src/rqt_launch/node_delegate.py:47]: Commented code?

File: src/rqt_launch/node_proxy.py
Comment (brawner) [inline src/rqt_launch/node_proxy.py]: Don't break the first line and then align second with str.format

File: src/rqt_launch/node_proxy.py
Comment (brawner) [inline src/rqt_launch/node_proxy.py:75]: Align with str.format

File: src/rqt_launch/node_proxy.py
Comment (brawner) [inline src/rqt_launch/node_proxy.py:79]: Commented code

File: src/rqt_launch/node_proxy.py
Comment (brawner) [inline src/rqt_launch/node_proxy.py:98]: Is it better to break at the =?

File: src/rqt_launch/node_widget.py
Comment (brawner) [inline src/rqt_launch/node_widget.py]: Don't break this line

File: src/rqt_launch/node_widget.py
Comment (brawner) [inline src/rqt_launch/node_widget.py]: Don't break line

File: src/rqt_launch/status_indicator.py
Comment (brawner) [inline src/rqt_launch/status_indicator.py]: Don't break line

File: src/rqt_launch/status_indicator.py
Comment (brawner) [inline src/rqt_launch/status_indicator.py]: Don't break line

File: src/rqt_launch/status_indicator.py
Comment (brawner) [inline src/rqt_launch/status_indicator.py]: Don't break line

File: src/rqt_launch/launch_widget.py
Comment (rsinnet) [reply src/rqt_launch/launch_widget.py]: This alignment is actually fine according to PEP 8: https://www.python.org/dev/peps/pep-0008/#indentation

File: src/rqt_launch/launch_widget.py
Comment (rsinnet) [reply src/rqt_launch/launch_widget.py]: Maybe it should be deleted in a separate PR. I didn't write this so I'm not sure but it would be worth looking through the history before removing commented out code. Hopefully, the author put a note about why it was commented out in the commit message. In addition, one deletion per commit with an explanation of why it can be removed and reference to they previous commit(s) where it was deprecated.

File: src/rqt_launch/node_controller.py
Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: The break is necessary to keep it under 79 characters as specified in PEP 8.

File: src/rqt_launch/node_controller.py
Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: The break is necessary to keep it under 79 characters as specified in PEP 8.

File: src/rqt_launch/node_controller.py
Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: The break is necessary to keep it under 79 characters as specified in PEP 8.

File: src/rqt_launch/node_proxy.py
Comment (rsinnet) [reply src/rqt_launch/node_proxy.py]: That won't work because it would exceed 79 characters,

**Review by rsinnet — APPROVED** [review]
I'll approve because I'm a fan of PEP 8 although autopep8 requests are usually a maintainer's nightmare :trollface: I think most of the comments from @brawner don't take into account maximum line length. Also, let's not remove any commented out code in this PR. Everything here seems to be to spec based on PEP 8.

File: src/rqt_launch/node_controller.py
Comment (brawner) [reply src/rqt_launch/node_controller.py]: Python guide for ROS1 doesn't have a line length, and defaults to ROS C++ guide which is 120. But this is a pull request for preparation for porting to ROS 2, which has a 100 character line length.

File: src/rqt_launch/launch_widget.py
Comment (brawner) [reply src/rqt_launch/launch_widget.py]: It may be fine, but it doesn't make it more readable

File: src/rqt_launch/node_controller.py
Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: If that is the case, then should we actually be applying autopep8 at all? Is there a configuration file for ROS 2?

File: src/rqt_launch/launch_widget.py
Comment (rsinnet) [reply src/rqt_launch/launch_widget.py]: That's a highly subjective claim. I actually think it's more readable :smile:

File: src/rqt_launch/node_controller.py
Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: Can you share a link to the style guide for ROS 1? I only know of this: http://wiki.ros.org/PyStyleGuide

That guide specifies PEP 8 but lists a few exceptions.

Is the goal to actually use this repo for the ROS 2 version of `rqt_launch` or to do a fork to avoid breaking backward compatibility. Based on the significant overhaul to the launch system in ROS 2, I would think we would want a fork which means we probably wouldn't want to apply the ROS 2 style guide to this ROS 1 package. Thoughts, @brawner?

Comment (brawner) [reply src/rqt_launch/node_controller.py]: I guess the PyStyleGuide doesn't really say what I was mentioning. I had just heard line lengths were adopted from C++ standards. If this package uses 79, then we should just stick with that.

For now, we'll be creating crystal-devel branches in the RQT packages we port to ROS2. But the packages that should be forked vs branched are not entirely resolved. So it's good to hear some comments about it.

Comment (rsinnet) [reply src/rqt_launch/node_controller.py]: Cool, let me know when this gets resolved, I'm happy to support the decision.

File: src/rqt_launch/launch_widget.py
Comment (mlautman) [reply src/rqt_launch/launch_widget.py]: Turns out the entire thing is less than 100 char so I'm making it one line. That way no one is happy :stuck_out_tongue_closed_eyes:

File: src/rqt_launch/node_proxy.py
Comment (mlautman) [reply src/rqt_launch/node_proxy.py:75]: I generally don't like doing that because it makes the indentation dependent on the string being formatted and every change to the string then requires we re-indent for a bunch of lines

File: src/rqt_launch/node_controller.py
Comment (130s) [reply src/rqt_launch/node_controller.py]: Thanks for this improvement!

I can't speak for ROS2 branching. That said, and while I'm afraid of being pedantic, [PEP8 defines 79 character length](https://www.python.org/dev/peps/pep-0008/#maximum-line-length), which I believe I tried to stick to as much as possible when I made this (experimental) package. Back then there was a discussion for the line length at Willow for Python code, which didn't seem to have ended up making change in the [wiki.ros.org/PyStyleGuide](http://wiki.ros.org/PyStyleGuide). So I'm +1 for 79 limit wherever that makes sense.

**Comment by rsinnet** [comment]
It's looking good to me. We can wait until we get resolution from stakeholders to merge (@brawner, @130s)

@mlautman, have you heard of the autosquash feature in Git? It will change your life 💪 
https://codegoalie.com/2013/04/04/better-pull-requests-with-git---autosquash

Based on what I've heard, it sounds like this is fine and the fork can happen after for ROS 2.

**Comment by mlautman** [comment]
Autosquash is great!

**Review by brawner — APPROVED** [review]

**Review by 130s — APPROVED** [review]

**Pull request merged**
